pkg/serverless/aws/arn.go

```diff
@@ -115,3 +160,89 @@ func RestoreCurrentStateFromFile() error {
 	SetRequestID(restoredState.CurrentReqID)
 	return nil
 }
+
+// BuildFunctionARNFromEnv reconstructs the function arn from what's available
+// in the environment.
+func BuildFunctionARNFromEnv() string {
+	region := os.Getenv(regionEnvVar)
+	functionName := os.Getenv(functionNameEnvVar)
+	alias := os.Getenv(aliasEnvVar)
+	accountID := fetchAccountID()
+	arnPrefix := "arn:aws"
+
+	if len(accountID) == 0 || len(region) == 0 || len(functionName) == 0 {
+		return ""
+	}
+
+	if strings.HasPrefix(region, "us-gov") {
+		arnPrefix = "arn:aws-us-gov"
+	}
+	if strings.HasPrefix(region, "cn-") {
+		arnPrefix = "arn:aws-cn"
+	}
+
+	baseARN := fmt.Sprintf("%s:lambda:%s:%s:function:%s", arnPrefix, region, accountID, functionName)
+	if len(alias) > 0 && alias != "$LATEST" {
+		return fmt.Sprintf("%s:%s", baseARN, alias)
```

Comment:
It looks like we are including the qualifier/version/alias in the ARN as well as storing it separately. I think this is different from the behavior we had before. Is that intentional?

Reply:
This is trying to replicate the invokedARN that is given on each invocation, which has the qualfiier, and then we save it using SetARN, which will separate the qualifier back out.